packages/myst-cli/src/transforms/embed.ts

```diff
@@ -35,13 +36,28 @@ function mutateEmbedNode(
   const { url, dataUrl, targetFile, sourceFile } = opts ?? {};
   if (targetNode && node['remove-output']) {
     targetNode = filter(targetNode, (n: GenericNode) => {
+      // After reduction, 'output' nodes may be replaced by their children which are then tagged as outputs
       return n.type !== 'output' && n.data?.type !== 'output';
     });
   }
   if (targetNode && node['remove-input']) {
-    targetNode = filter(targetNode, (n: GenericNode) => {
-      return n.type !== 'code' || n.data?.type === 'output';
-    });
+    targetNode = filter(
+      targetNode,
+      (
+        n: GenericNode,
+        index: number | null | undefined,
+        parent: GenericNode | undefined | null,
+      ) => {
+        // Code nodes that are children of notebook cells and _not_ tagged as outputs
+        // should be removed
+        return !(
+          n.type === 'code' &&
+          n.data?.type !== 'output' &&
```

Comment:
@fwkoch I don't think I need this branch actually, because `reduceOutputs` is always called _after_ this transform runs? So there should never be such data in the file.